script.js

```diff
@@ -1,4 +1,283 @@
+        cardName = `King`;
+      }
+
+      //create a new card object with name, suit, rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        emoji: emoji[suitIndex],
+      };
+
+      //change jack, queen, king to rank 10
+      if (card.rank == 11 || card.rank == 12 || card.rank == 13) {
+        card.rank = 10;
+      }
+
+      //add card to the deck
+      deck.push(card);
+    }
+  }
+  return deck;
+};
+
+var deck = makeDeck();
+
+///shuffle
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+// setting rank as per black jack rules
+var counter = 0;
+while (counter < deck.length) {
+  if (
+    deck[counter].name == "jack" ||
+    deck[counter].name == "queen" ||
+    deck[counter].name == "king"
+  ) {
+    deck[counter].rank = 10;
+  }
+
+  counter = counter + 1;
+}
+
+// shuffle deck
+var shuffledDeck = shuffleCards(deck);
+
+// game modes  input name , player hand  ,
+var gameMode = "inputName";
+var playerName = "";
+var yourBetAmount = 0;
+var playerScore = 0;
+var card1 = {};
+var card2 = {};
+var dealerCards = [];
+
+var printCard = function (card) {
+  cardPic = `<div id="blackcard">${card.name}<br><span class="emoji">${card.emoji}</span></div>`;
+
+  if (card.suit == "hearts" || card.suit == "diamonds") {
+    cardPic = `<div id="redcard">${card.name}<br><span class="emoji">${card.emoji}</span></div>`;
+  }
+
+  return cardPic;
+};
+
+// modes inputName , betAmount, deal2cards , decidehit/Stay , dealerMode 2card autohit
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "inputName") {
+    playerName = input;
+    gameMode = "betAmount";
+    return (
+      "🥂 Hi " +
+      playerName +
+      ", welcome to the 'No Limit Blackjack Table'" +
+      "<br>" +
+      "👉 Please type in your betting amount 💰💰💰 in the orange bar above !"
+    );
+  }
+  if (gameMode == "betAmount") {
+    yourBetAmount = input;
+    gameMode = "ace1/11";
+    return (
+      "🥂 Great " +
+      playerName +
+      ", today seems to be your lucky day, 💰 you have bet: " +
+      yourBetAmount +
+      "<br>" +
+      "🤞 You can decide the Rank of Ace to be either 1 or 11" +
+      "<br>" +
+      "👉 Type in your choice: 1 or 11 in the orange bar above."
+    );
+  }
+
+  if (gameMode == "ace1/11") {
+    gameMode = "dealCards";
+    var counter = 0;
+    while (counter < shuffledDeck.length) {
+      if (shuffledDeck[counter].name == "ace") {
+        shuffledDeck[counter].rank == input;
```

Comment:
I realised that your input isn't properly setting the Aces to 1 or 11. Did you notice a typo in this line here that causes the assignment not to be accurate?